@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -0,0 +1,524 @@
+        readonly nearMultiplier: number = 0.1,
+        readonly nearFarMultiplier = 50.0,
+        readonly farOffset = 200.0
+    ) {
+        assert(nearMin > 0);
+        assert(nearFarMultiplier >= 0);
+        assert(farOffset >= 0);
+        this.farMin = nearMin * nearFarMultiplier + farOffset;
+    }
+
+    evaluateClipPlanes(
+        camera: THREE.Camera,
+        projection: Projection
+    ): { near: number; far: number } {
+        let nearPlane: number = this.nearMin;
+        let farPlane: number = this.farMin;
+        if (projection.type === ProjectionType.Spherical) {
+            // near and far plane for a set up where
+            // the camera is looking at the center of the scene.
+            const r = EarthConstants.EQUATORIAL_RADIUS;
+            const d = camera.position.length();
+            const alpha = Math.asin(r / d);
+            // Extract X, Y, Z axes into tmp vectors array.
+            camera.matrixWorld.extractBasis(
+                this.m_tmpVectors[0],
+                this.m_tmpVectors[1],
+                this.m_tmpVectors[2]
+            );
+            // Setup quaternion based on X axis.
+            this.m_tmpQuaternion.setFromAxisAngle(this.m_tmpVectors[0], alpha);
+            // Acquire forward vector based on Z axis reversed (keep it in tmpVectors[2]).
+            const fwd = this.m_tmpVectors[2].negate();
+            // Apply quaternion rotation to forward vector, store it in tmpVectors[1].
+            const fwdRot = this.m_tmpVectors[1].copy(fwd).applyQuaternion(this.m_tmpQuaternion);
+            // Store camera position tmpVectors[0] and reference it with p.
+            const p = this.m_tmpVectors[0].copy(camera.position);
+            p.addScaledVector(fwdRot, Math.sqrt(d * d - r * r));
+            farPlane = p.sub(camera.position).dot(fwd);
+            const bias = 2000; // TODO: generalize.
+            nearPlane = Math.max(this.nearMin, projection.groundDistance(camera.position) - bias);
+        } else {
+            const groundDistance = projection.groundDistance(camera.position);
+            nearPlane = Math.max(this.nearMin, groundDistance * this.nearMultiplier);
+            // Will be already clamped to minFar due to clamping above.
+            farPlane = nearPlane * this.nearFarMultiplier + this.farOffset;
+        }
+        return { near: nearPlane, far: farPlane };
+    }
+}
+
+/**
+ * Simple camera clip planes evaluator that computes near plance based on camera to ground distance.
+ *
+ * This evaluator supports both planar and spherical projections, although it behaviour is
+ * slightly different in each case. General algorithm setups near plane and far plane close
+ * to ground level, but taking into account some predefined margins.
+ * @note This simple evaluator does behave correctly when camera is always pointing downwards the
+ * ground surface (top view) along surface normal, but does not preserve correct planes when using
```

Comment:
Does it make sense to have an evaluator that doesn't support tilt?

Reply:
Yes, for top view maps (2d like) and it may be (and is) used in examples.

Reply:
Which example uses this?

Reply:
All examples that utilizes MapView use InterpolatedClipPlanesEvaluator, but the idea is to get rid of this and provide altitude based approach there. For more sophisticated stuff (map view demo) would like to use tilt based approach.